drivers/client_location_history/app/controllers/client_location_history/projects_controller.rb

```diff
@@ -0,0 +1,60 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module ClientLocationHistory
+  class ProjectsController < ApplicationController
+    before_action :set_project
+
+    def map
+      @locations = @project.enrollment_location_histories.where(located_on: filter.range)
+      @markers = @locations.map { |l| l.as_marker(current_user, [:name, :seen_on]) }
+      @bounds = ClientLocationHistory::Location.bounds(@locations)
+      @options = {
+        bounds: @bounds,
+        cluster: true,
+        marker_color: ClientLocationHistory::Location::MARKER_COLOR,
+      }
+    end
+
+    private def set_project
+      @project = project_scope.includes(:enrollment_location_histories).find(params[:id].to_i)
```

Comment:
I tested this out locally with a user who shouldn't have permission
- Doesn't have any permissions on this project: works fine, they are unable to view
- Has view permissions on this project, but not `can_view_project_locations` - can see even though should not. in the debugger I can see that `using_acls?` is false for this user, but I don't know how to turn that on.

Reply:
In the case where you aren't using ACLs, this should return data if you have access to the project through any group and any role with `can_view_project_locations`.  If you are using ACLs (set on the user edit screen), then you should need an access control that has a role with `can_view_project_locations` AND a collection containing the project.

Reply:
Ok, thanks, this was a very helpful introduction to the warehouse legacy permissions! I tested and think everything looks good now. I re-added the `can?` permission on project that we discussed earlier, in order to granularly determine whether the link to the project map should be shown.

Tested:

Not using ACLs
- User who has access to the project through any group, and has a role granting both `can_view_projects` and `can_view_project_locations`
  - Can see link on project page
  - Can see map
- User who has access to the project but has a role only granting `can_view_projects`, not `can_view_project_locations`
  - Cannot see link on project page
  - If they try to navigate there directly, they get "Sorry you are not authorized to do that"

Using ACLs
- User who has `can_view_project` but not `can_view_project_locations` on this project, even though they may have `can_view_project_locations` on another project:
  - Can't see the link on the project page
  - If they navigate directly to the route, they get `ActiveRecord::RecordNotFound`
- User who has `can_view_project` and `can_view_project_locations`:
  - Can see link on project page
  - Can see map
- User who has `can_view_project_locations` and not `can_view_project` (This should not happen)
  - Cannot see project page at all
  - If they navigate directly to the route, they get `ActiveRecord::RecordNotFound`

Reply:
That sounds perfect.  Thanks @martha !

Reply:
Closing the circle here for posterity: There was a bug with the case `User who has access to the project but has a role only granting can_view_projects, not can_view_project_locations`. I was fooled by local caching when I tested it and assumed it was working. See PR fix: https://github.com/greenriver/hmis-warehouse/pull/4787